test/functional/mempool_accept.py

```diff
@@ -159,64 +139,53 @@ def run_test(self):
-        txid_1 = node.sendrawtransaction(hexstring=raw_tx_1, maxfeerate=0)
-        # Now spend both to "clearly hide" the outputs, ie. remove the coins from the utxo set by spending them
-        raw_tx_spend_both = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[
-                {'txid': txid_0, 'vout': 0},
-                {'txid': txid_1, 'vout': 0},
-            ],
-            outputs=[{node.getnewaddress(): 0.1}]
-        ))['hex']
-        txid_spend_both = node.sendrawtransaction(hexstring=raw_tx_spend_both, maxfeerate=0)
+        node.sendrawtransaction(hexstring=tx_0['hex'], maxfeerate=0) # spend the tx_0
+        self.mempool_size += 1
+        tx_1 = miniwallet.send_self_transfer(fee_rate=4*fee_rate,from_node=node)
+        utxo_1 = miniwallet.get_utxo()
+        miniwallet.send_self_transfer(from_node=node, utxo_to_spend=utxo_1)['txid']
         node.generate(1)
         self.mempool_size = 0
+
         # Now see if we can add the coins back to the utxo set by sending the exact txs again
         self.check_mempool_result(
-            result_expected=[{'txid': txid_0, 'allowed': False, 'reject-reason': 'missing-inputs'}],
-            rawtxs=[raw_tx_0],
+            result_expected=[{'txid': tx.rehash(), 'allowed': False, 'reject-reason': 'missing-inputs'}],
+            rawtxs=[tx.serialize().hex()],
         )
         self.check_mempool_result(
-            result_expected=[{'txid': txid_1, 'allowed': False, 'reject-reason': 'missing-inputs'}],
-            rawtxs=[raw_tx_1],
+            result_expected=[{'txid': tx_1['txid'], 'allowed': False, 'reject-reason': 'missing-inputs'}],
+            rawtxs=[tx_1['hex']],
         )
 
         self.log.info('Create a signed "reference" tx for later use')
-        raw_tx_reference = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': txid_spend_both, 'vout': 0}],
-            outputs=[{node.getnewaddress(): 0.05}],
-        ))['hex']
-        tx = tx_from_hex(raw_tx_reference)
+        utxo_2 = miniwallet.get_utxo()
+        tx_reference = miniwallet.create_self_transfer(from_node=node, utxo_to_spend=utxo_2)
         # Reference tx should be valid on itself
         self.check_mempool_result(
-            result_expected=[{'txid': tx.rehash(), 'allowed': True, 'vsize': tx.get_vsize(), 'fees': { 'base': Decimal('0.1') - Decimal('0.05')}}],
-            rawtxs=[tx.serialize().hex()],
+            result_expected=[{'txid': tx_reference['txid'], 'allowed': True, 'vsize': tx_reference['tx'].get_vsize(), 'fees': {'base': tx_reference['fees']}}],
+            rawtxs=[tx_reference['hex']],
             maxfeerate=0,
         )
 
         self.log.info('A transaction with no outputs')
-        tx = tx_from_hex(raw_tx_reference)
+        tx = tx_from_hex(tx_reference['hex'])
         tx.vout = []
         # Skip re-signing the transaction for context independent checks from now on
```

Comment:
Can also remove this line, because there are no signatures

Reply:
`tx.vout = []` is necessary since the transaction is serialized in `check_mempool_result`. Without this, the transaction does not get rejected for the right reasons. Please let me know if I misunderstood the suggestion.

Reply:
I meant the comment line :sweat_smile: